jetty-server/src/main/java/org/eclipse/jetty/server/HttpInput.java

```diff
@@ -524,6 +524,7 @@ protected void skip(Content content, int length)
      */
     protected void blockForContent() throws IOException
     {
+        assert _lock.isHeldByCurrentThread();
```

Comment:
interesting.  is there a reason for this here?

Reply:
Yes, few lines below there is a naked call to `_lock.await()` which can only happen if the caller of `blockForContent()` is holding the lock. The assert makes explicit that we want `blockForContent()` only ever called with the lock held.